storage/store.go

```diff
@@ -900,7 +900,7 @@ func (s *Store) RemoveRange(rng *Range) error {
 	n := sort.Search(len(s.rangesByKey), func(i int) bool {
 		return bytes.Compare(rng.Desc().StartKey, s.rangesByKey[i].Desc().EndKey) < 0
 	})
-	if n >= len(s.rangesByKey) {
+	if n >= len(s.rangesByKey) || rng.Desc().RaftID != s.rangesByKey[n].Desc().RaftID {
```

Comment:
@kkaneda Feel free to do the investigation yourself if you'd like. My thought was to compare insertion and lookup for 1M ranges using a sorted slice vs llrb vs a btree (e.g. https://github.com/google/btree). Whichever looks best we should use for both `RangeCache` and here.